client/data/newsletter-categories/use-set-newsletter-category-mutation.tsx

```diff
@@ -0,0 +1,47 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type useSetNewsletterCategoryMutationParams = {
+	siteId: number;
+};
+
+type SetNewsletterCategoryParams = {
+	id: number;
+};
+
+type SetNewsletterCategoryResponse = {
+	success: boolean;
+};
+
+const useSetNewsletterCategoryMutation = ( { siteId }: useSetNewsletterCategoryMutationParams ) => {
+	const queryClient = useQueryClient();
+	const cacheKey = [ `newsletter-categories-${ siteId }` ];
+	return useMutation( {
+		mutationFn: async ( { id }: SetNewsletterCategoryParams ) => {
```

Comment:
I have the same thought here: why not just `(id: number) `? Chances of this function needing additional parameters seem slim.